frontend/app/services/cardService.ts

```diff
@@ -1,38 +1,39 @@
 import { CardData } from '../types/card';
 
 // Constants
-const API_DELAY = 500; // Simulated API delay in milliseconds
 export const CARDS_PER_PAGE = 20;
-
-// Helper function to generate mock card data
-const generateMockCard = (id: number): CardData => ({
-  id,
-  title: `Card ${id}`,
-  content: `This is the content of card ${id}. You can add more information here.`,
-});
-
-// Helper function to simulate API delay
-const simulateApiDelay = () =>
-  new Promise((resolve) => setTimeout(resolve, API_DELAY));
+const BASE_URL = 'http://localhost:8000/idioms/';
 
 /**
- * Fetches a paginated list of cards
+ * Fetches a paginated list of cards from the backend
  * @param page - The page number to fetch (1-based)
  * @param limit - Number of cards per page
+ * @param search - Optional search query
  * @returns Promise with array of cards
  */
 export const fetchCards = async (
   page: number,
   limit: number = CARDS_PER_PAGE,
+  search?: string,
 ): Promise<CardData[]> => {
-  await simulateApiDelay();
+  try {
+    const url = new URL(BASE_URL);
+    url.searchParams.append('page', page.toString());
+    url.searchParams.append('limit', limit.toString());
+    if (search) {
+      url.searchParams.append('search', search.trim());
+    }
```

Comment:
add a console log here that this is not currently supported, otherwise the backend might return an error.